MatrixTest.java

```diff
@@ -0,0 +1,163 @@
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotSame;
+import org.junit.Test;
+
+
+public class MatrixTest{
+    //Matrix(double[][] A){}
+
+    //Matrix(double[][] A, int m, int n){}
+    @Test
+    public void baseCtorSameLength(){
+	double[][] array = {
+		{1,2,3},
+		{4,5,6},
+		{7,8,9}
+	};
+	Matrix m = new Matrix(array, 3, 3);
+	double[][] array2 = m.getArray();
+        assertEquals(array, array2);
+
+    }
+    @Test
+    public void baseCtorNotSameObj(){
```

Comment:
Can we always write a one-sentence comment for our functions. I think checkstyle requires it, and it also helps understand the intention of a test before you start reading the code.